File: backend/src/main/java/com/sistemaOficina/backend/entidade/Veiculo.java
package com.sistemaOficina.backend.entidade;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Veiculo {
//Veiculo(placa, quilometragem, chassi, patrimonio, ano_modelo, ano_fabricacao, id_modelo) id_modelo referencia Modelo(id)

    private Long id;
    private String placa;
    private int quilometragem;
    private String chassi;
    private String patrimonio;
    private int anoModelo;
    private int anoFabricacao;
    private Modelo idModelo;
    private AcessorioVeiculo idAcessorioVeiculo;
    private OrdemDeServico  idOrdemDeServico;
    private Propriedade idPropriedade;

}

File: frontend/src/app/models/Veiculo.ts
import { Modelo } from "./Modelo";

export class Veiculo {
  id?: number;
  placa?: string;
  quilometragem?: number;
  chassi?: string;
  patrimonio?: string;
  anoModelo?: number;
  anoFabricacao?: number;
  idModelo?: Modelo;
}
